internal/services/loganalytics/migration/saved_search_v0_to_v1.go

```diff
@@ -0,0 +1,96 @@
+package migration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics/parse"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+)
+
+var _ pluginsdk.StateUpgrade = SavedSearchV0ToV1{}
+
+type SavedSearchV0ToV1 struct{}
+
+func (SavedSearchV0ToV1) Schema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"log_analytics_workspace_id": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			ValidateFunc:     validate.LogAnalyticsWorkspaceID,
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		"name": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Same for these as well as for the properties below.

Reply:
it's a state migration, so it's worth leaving this in for now